packages/admin/admin/src/error/errordialog/ErrorDialogProvider.tsx

```diff
@@ -1,23 +1,24 @@
+import { useReactiveVar } from "@apollo/client";
 import * as React from "react";
 
 import { ErrorDialog, ErrorDialogOptions } from "./ErrorDialog";
-import { ErrorDialogContext, ErrorDialogContextProps } from "./ErrorDialogContext";
+import { errorDialogVar } from "./errorDialogVar";
 
 export const ErrorDialogProvider: React.FunctionComponent = ({ children }) => {
-    const [errorOptions, setErrorOptions] = React.useState<ErrorDialogOptions | undefined>(undefined);
+    const errorOptions = useReactiveVar<ErrorDialogOptions | undefined>(errorDialogVar);
+
     const [errorDialogVisible, setErrorDialogVisible] = React.useState(false);
 
-    const errorDialog = React.useMemo((): ErrorDialogContextProps => {
-        return {
-            showError: (options: ErrorDialogOptions) => {
-                setErrorOptions(options);
-                setErrorDialogVisible(true);
-            },
-        };
-    }, []);
+    React.useEffect(() => {
+        if (errorOptions && !errorDialogVisible) {
+            setErrorDialogVisible(true);
+        } else if (errorOptions == null && errorDialogVisible) {
+            setErrorDialogVisible(false);
+        }
+    }, [errorDialogVisible, errorOptions]);
```

Comment:
I believe this useEffect can be greatly simplified:

```typescript
React.useEffect(() => {
    setErrorDialogVisible(errorOptions != null);
}, [errorOptions]);
```

Why would you need to check whether the error dialog is already visible? And setting the `errorDialogVisible` inside a useEffect that has `errorDialogVisible` as dependency may lead to infinite updates?

Reply:
true, thought tooooo complicated :-)